ci/check-protobuf-stability.sh

```diff
@@ -35,42 +49,65 @@ function check_non_lf_protos() {
   trap 'rm -rf ${BUF_IMAGE_TMPDIR}' EXIT
 
-  # The `SYSTEM_PULLREQUEST_TARGETBRANCH` environment variable is defined by
-  # Azure Pipelines; in order to run this script locally, define it beforehand
-  # as the branch being targeted. For example:
-  #
-  #   SYSTEM_PULLREQUEST_TARGETBRANCH=main bash -x ci/check-protobuf-stability.sh
-  #
-  TARGET="${SYSTEM_PULLREQUEST_TARGETBRANCH:-main}"
-  echo "The target branch is '${TARGET}'."
+  echo "Checking protobufs against target '${BUF_TAG_TO_CHECK}'"
+  eval "$(dev-env/bin/dade assist)"
+  for buf_module in "${BUF_MODULES_AGAINST_STABLE[@]}"; do
+    # Starting with version 1.17 we split the default `buf.yaml` file into multiple config files
+    # This in turns requires that we pass the `--against-config` flag for any check that is run on versions > 1.17
+    if [[ $BUF_CONFIG_UPDATED ]]
+    then
+      buf breaking --config "${buf_module}" --against "$BUF_TAG_TO_CHECK" --against-config "${buf_module}"
+    else
+      buf breaking --config "${buf_module}" --against "$BUF_TAG_TO_CHECK"
+    fi
+  done
 
-  # For `main` and PRs targeting `main`, we simply check against the most recent
-  # stable tag.
-  #
-  # For PRs targeting release branches, we should really check against
-  # all the most recent stable tags reachable from either the current branch or
-  # from previous release branches (say, against both `1.17.1` and `1.16.2`
-  # created after the `release/1.17.x` branch).
-  # Instead, we approximate by checking only against the most recent stable tag
-  # reachable from the current branch, under the assumption that if a lesser
-  # release branch contains a protobuf change, then it will also be present in
-  # higher ones either through a shared commit or a back-port from `main`.
+}
+
+case "${1:---stable}" in
+-h | --help)
+  cat <<USAGE
+Usage: ./check-protobuf-stability.sh [options]
+
+Options:
+  -h, --help: shows this help
+  --stable:     check against the latest stable tag (default)
+  --head:       check against the head commit
+USAGE
+  exit
+  ;;
+--stable)
+  echo "Using stable target"
+  # Check against the most recent stable tag.
   #
-  # Finally, this check does not need to run on release branch commits because
+  # This check does not need to run on release branch commits because
   # they are built sequentially, so no conflicts are possible and the per-PR
   # check is enough.
+  readonly RELEASE_BRANCH_REGEX="^release/.*"
   GIT_TAG_SCOPE=""
   if [[ "${TARGET}" =~ ${RELEASE_BRANCH_REGEX} ]]; then
     GIT_TAG_SCOPE="--merged"
   fi
-
   readonly LATEST_STABLE_TAG="$(git tag ${GIT_TAG_SCOPE} | grep -v "snapshot" | sort -V | tail -1)"
-  echo "Checking protobufs against tag '${LATEST_STABLE_TAG}'"
-  for buf_module in "${BUF_MODULES_AGAINST_STABLE[@]}"; do
-    (eval "$(dev-env/bin/dade assist)" ; buf breaking --config "${buf_module}" --against ".git#tag=${LATEST_STABLE_TAG}")
-  done
-
-}
-
+  # For the latest stable snapshot the config is still the default "buf.yaml"
```

Comment:
Do you mean with "stable release"? Will this require a further change after 1.18? Isn't it better to check for the specific release so that we just have to remove legacy code, rather than fixing the build?

Reply:
No, I misworded this one. It should've been actually `latest stable tag`

Reply:
Good point. I will add a story to remove the check after the 1.18 release, and will update the code to check the release version